harmonica/meshes.py

```diff
@@ -0,0 +1,212 @@
+    region,
+    spacing=None,
+    shape=None,
+    adjust="spacing",
+    top=None,
+    bottom=None,
+    density=None,
+    coordinates="spherical",
+    region_centers=True,
+):
+    """
+    Create a layer of tesseroids.
+
+    The layer can be specified by either the number of tesseroids in each dimension (the
+    *shape*) or by the size of each tesseroid (the *spacing*).
+
+    The layer will be a :class:`xarray.Dataset` with ``longitude`` and ``latitude``
+    coordinates and `top`, `bottom` and `density` as data.
+    The ``longitude`` and ``latitude`` will be the coordinates of the center of each
+    tesseroid on the layer in geocentric spherical coordinates.
+    The ``top`` and ``bottom`` parameters can be defined either in a geocentric
+    spherical or in a geodetic coordinate system.
+
+    .. warning :
+        If ``top`` and ``bottom`` parameters are defined under a geodetic coordinate
+        system, they will be converted to a spherical geocentric one.
+        Because tesseroids cannot be defined on a geodetic coordinate system, the
+        latitude coordinates will remain unchanged.
+
+    If the given region is not divisible by the desired spacing, either the region or
+    the spacing will have to be adjusted. By default, the spacing will be rounded to the
+    nearest multiple. Optionally, the East and North boundaries of the region can be
+    adjusted to fit the exact spacing given. See the examples below.
+
+    Parameters
+    ----------
+    region : list = [W, E, S, N]
+        The boundaries of a given region in geocentric spherical coordinates in degrees.
+    spacing : float, tuple = (d_lat, d_lon), or None
+        The latitudinal and longitudinal spacing (in degrees) between the center of
+        neighbour tesseroids, respectively, i.e. the longitudinal and latitudinal size
+        of each tesseroid. A single value means that the spacing is equal in both
+        directions.
+    shape : tuple = (n_lat, n_lon) or None
+        The number of tesseroids in the latitudinal and longitudinal directions,
+        respectively.
+    adjust : {'spacing', 'region'}
+        Whether to adjust the spacing or the region if required. Ignored if *shape* is
+        given instead of *spacing*. Defaults to adjusting the spacing.
+    top : float or None
+        Coordinate (in meters) of outer surface of the layer in geocentric spherical or
+        geodetic coordinates. If ``None`` a ``np.nan`` array will be added to the
+        :class:``xarray.Dataset``.
+    bottom : float or None
+        Coordinate (in meters) of inner surface of the layer in geocentric spherical or
+        geodetic coordinates. If ``None`` a ``np.nan`` array will be added to the
+        :class:``xarray.Dataset``.
+    density : float or None
+        Density (in SI units) of the tesseroids on the layer. If ``None`` a zeroes array
+        will be added to the :class:``xarray.Dataset``.
+    coordinates : {"spherical", "geodetic"}
```

Comment:
What if instead of this, we take `projection = None or function`. In this case, projection would be any function that transforms coordinates into a spherical geocentric system.

Reply:
I like that!
With this in mind, maybe we could create a private function to create `layers` or `meshes` and then call it for each specific geometry (tesseorid, prisms, etc).
So, when using prisms, the projection could be one from `pyproj`, while on the tesseroid case, it could be `geodetic_to_spherical`.